bin/add_video_tags.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Namratha Urs <namrathaurs@my.unt.edu>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This script is used to add video tags to the Anthology towards ingestion of videos.
+
+Usage:
+    add_video_tags.py TSV_files
+
+        where TSV_files are the tab-separated values (TSV) files containing the tuples (anthology_id, presentation_id)
+
+    Consolidates all the TSV files passed to the script, edits the XML by adding a properly-indented video tag to the
+    end of the <paper> element and rewrites the XML.
+"""
+
+import pandas as pd
+import os, glob
+import lxml.etree as ET
+import argparse
+
+from anthology.utils import (
+    deconstruct_anthology_id,
+    make_simple_element,
+    indent
+)
+
+script_root = os.path.dirname(os.path.abspath(__file__))
+data_dir = os.path.join(script_root, "../data/xml")
+
+def combine_tsv(files):
+    combined_df = pd.concat([pd.read_csv(os.path.join(script_root, fname), sep="\t") for fname in files])
+    return combined_df
+
+def split_anth_id(id):
+    coll_id, _, _ = deconstruct_anthology_id(id)
+    return coll_id
+
+def add_video_tag(anth_paper, xml_file):
+    coll_id, vol_id, paper_id = deconstruct_anthology_id(anth_paper.anthology_id)
+    tree = ET.parse(os.path.join(data_dir, xml_file))
+
+    paper = tree.getroot().find(f'./volume[@id="{vol_id}"]/paper[@id="{paper_id}"]')
+    video_url = "http://slideslive.com/{}".format(anth_paper.presentation_id)
+    make_simple_element("video", attrib={"tag": "video", "href": video_url}, parent=paper)
+
+    with open(os.path.join(data_dir, coll_id + ".xml"), 'wb') as f:
+        tree.write(f, encoding="UTF-8", xml_declaration=True)
+
+
+def main(args):
+    combo_df = combine_tsv(args['tsv_files'])
+    combo_df_uniques = combo_df['anthology_id'].apply(split_anth_id).unique()
+
+    for xml in os.listdir(data_dir):
+        fname, ext = os.path.splitext(xml)
+        if fname in combo_df_uniques.tolist() or fname == "2020.acl":
```

Comment:
Currently, you are opening, parsing, and writing the XML file for every paper. I suggest opening the XML file here, getting the `paper` node for each paper, and passing it into `add_video_tag()`. It should run much faster then.

Reply:
This suggestion has been incorporated in commit 01588a9.